packages/runtime/runtime-definitions/src/summary.ts

```diff
@@ -23,16 +38,25 @@ export interface ISummaryTreeWithStats {
     summary: ISummaryTree;
 }
 
+export interface IChannelSummarizeResult extends ISummaryTreeWithStats {
+    references: IFluidObjectReferences[];
+}
+
 export interface ISummarizeResult {
     stats: ISummaryStats;
     summary: SummaryTree;
 }
 
-export interface ISummarizeInternalResult extends ISummarizeResult {
```

Comment:
ISummaryTreeWithStats  and ISummarizeResult  are the same
hence IChannelSummarizeResult  & IContextSummarizeResult  are the same.
Can we merge them?

Reply:
ISummaryTreeWithStats and ISummarizeResult differ in that ISummarizeResult's summary can also be a ISummaryHandle. ISummaryTreeWithStats's summary is always an ISummaryTree.
We can probably remove ISummaryTreeWithStats and assert that channel's always return an ISummaryTree and never an ISummaryHandle.

Reply:
I will speak to @arinwt about this and consolidate this if possible in a separate PR.

Reply:
I like differing types, but think we could name them better

Reply:
Do you have any names in mind?

Reply:
No, I think those are fine. I meant the underlying type names are bad at distinguishing themselves: `ISummaryTreeWithStats` and `ISummarizeResult`.

Reply:
I think it boils down to `SummaryTree` being poorly named.
If we had a unique term for what `SummaryTree` currently is: `ISummaryTree | ISummaryHandle`. Then we could use that to distinguish all the way up the stack. For example:
`ISummaryTree` vs. `ISummaryTreeOrHandle`
`ISummaryTreeWithStats` vs. `ISummaryTreeOrHandleWithStats`
etc.
But `ISummaryTreeOrHandle` is pretty long...

Reply:
I agree. SummaryTree and ISummaryTree sound very similar and its strange how different they are :-).
Created issue #4434 to track this. Assigned to both you and me. One of us can take this up.